### PR TITLE
chore: Bump holidays to 0.28

### PR DESCRIPTION
**Pull request by arkid15r**
### SUMMARY

Upgrade `holidays` to v0.28.
Set holidays package upper bound to '<1.0'

### ADDITIONAL INFORMATION

- [x] Has associated issue: related to but may not fix a root cause of #24619 
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by EugeneTorap** [comment]
Thank you!
@dpgaspar @villebro Can you review it?

File: setup.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def get_git_sha() -> str:
         "geopy",
         "gunicorn>=20.1.0; sys_platform != 'win32'",
         "hashids>=1.3.1, <2",
-        "holidays>=0.23, <0.24",
+        "holidays>=0.28, <0.29",
```
Comment (john-bodley) [inline setup.py:95]: Typically the upper bound should be exclusive of the next major version (assuming it follows semver notation).

Comment (EugeneTorap) [reply setup.py:95]: @arkid15r Do you have the major version of `holidays`? If no, we can remove the upper bound or use `<1.0`

Comment (arkid15r) [reply setup.py:95]: I believe relaxing the upper bound a bit would be the best case scenario here. I'll update the PR to use `holidays>=0.28, <1.0`

@john-bodley thank you  for the suggestion.

**Comment by EugeneTorap** [comment]
Hi @rusackas @john-bodley! Merge it?

**Review by john-bodley — APPROVED** [review]

**Pull request merged**